### ISSUE TITLE
Unknown page fault bugs for log crate

### ISSUE THREAD

**Issue by coolyjg** [syswonder/ruxos#78]
It seems that current dev branch has unknown page fault bus:
- When running python with `net` feature, `info` will cause page fault. (without latest `mmap` implementation)
- When using latest mmap, Nginx with `debug` and `error` will cause page fault. 

Please give more information about theses bugs @ken4647 , @MrRobertYuan

**Comment by ken4647** [comment]
This bug may be caused by the neno register group not being saved when the exception occurs. The following code can reproduce the problem:
```c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <sys/mman.h>

void neon_example(char *src, char *dst) {
    asm volatile (
        "ld1 {v0.16b}, [%0]\n"  
        : 
        : "r" (src)  
        : "v0"  
    );

    asm volatile (
        "st1 {v0.16b}, [%0]\n"  
        : 
        : "r" (dst)  
        : "memory", "v0" 
    );
}

int main()
{
    char orginal[16] = "hello RuxOS!";
    char* dst_buffer = (char*)mmap(NULL,1024*1024,PROT_READ|PROT_WRITE, MAP_ANONYMOUS|MAP_SHARED, -1, 0);
    printf("target reisgter src=%p, dst=%p\n",orginal,dst_buffer);


    neon_example(orginal, dst_buffer);
    printf("%s\n", dst_buffer);
    for(size_t i=0; i<16;i++){
        printf("%x ",dst_buffer[i]);
    }
    printf("\n");
    return 0;
}
```
It can be found that when the register is not saved on site, a page fault occurs when using q0 (that is, v0) to transmit memory data, which will cause the original data in q0 to be set to zero.This can be avoided by saving the neno register when an exception occurs.

**Comment by coolyjg** [comment]
I think this issue is fixed by #82
